File: 1/calc-2.js
const fs = require("fs");
const { reduce, sortBy } = require("lodash");

const input = fs.readFileSync("./input-1.txt").toString();

const grouped = reduce(
  input.split("\n").slice(0, -1),
  (acc, n) => {
    // console.log({ acc, n });
    if (n === "") return [...acc, []];
    return [...acc.slice(0, -1), [...(acc.slice(-1)[0] ?? []), parseInt(n)]];
  },
  []
);

const sums = grouped.map((g) => g.reduce((acc, n) => acc + n, 0));

const sorted = sortBy(sums).reverse();

const max3 = sorted.slice(0, 3);

const sum = max3.reduce((acc, n) => acc + n, 0);

console.log(sum);

File: 8/calc-2.clj
(defn parse [xs]
  (map #(Integer/parseInt %) xs))

(let [input (slurp "input-1.txt")]
  (as-> input v
    (str/split v #"\n\n") ;; split into groups
    (map #(str/split % #"\n") v) ;; split into lines
    ;; (first v)
    (map parse v) ;; string to int
    (map #(reduce + %) v) ;; sum up
    (sort v)
    (reverse v)
    (take 3 v)
    (reduce + v)
    (println v)))
